# Homework 4: Task 2 "Fire Perimeter" 11/24/2024

### Section 2 Criteria:

2. Fire perimeter data retrieval and selection
Explore the data and write a brief summary of the information you obtained from the preliminary information. Your summary should include the CRS of the data and whether this is projected or geographic.

3. From your fire perimeter data, select the Thomas Fire boundary. The fire occurred in 2017.

4. Save only the 2017 Thomas Fire boundary as a geospatial file in the format of your choosing. The file should go into the data/ directory in your repository.

5. In a markdown cell, briefly explain your reasoning for selecting that specific file format.



In [1]:
# Importing the libraries for this project
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import os 
import geopandas as gpd
import pyproj

In [2]:
# Creating a filepath to the data, and reading it in using geopandas
fp= os.path.join('data','California_Fire_Perimeters_7897608464518632307.geojson')
fire_perimeters= gpd.read_file(fp)

In [3]:
# Taking a look at the number of rows and columns in the GeoPandas Data Frame
fire_perimeters.shape

(22261, 20)

In [4]:
# Looking at the first 5 rows of the gpdDataframe
fire_perimeters.head()

OBJECTID   YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  \
0         1  2023.0    CA    CDF     SKU  WHITWORTH  00004808   
1         2  2023.0    CA    LRA     BTU     KAISER  00010225   
2         3  2023.0    CA    CDF     AEU    JACKSON  00017640   
3         4  2023.0    CA    CDF     AEU     CARBON  00018821   
4         5  2023.0    CA    CDF     AEU    LIBERTY  00018876   

                      ALARM_DATE                      CONT_DATE  CAUSE  \
0  Sat, 17 Jun 2023 00:00:00 GMT  Sat, 17 Jun 2023 00:00:00 GMT    5.0   
1  Fri, 02 Jun 2023 00:00:00 GMT  Fri, 02 Jun 2023 00:00:00 GMT    5.0   
2  Sat, 01 Jul 2023 00:00:00 GMT  Sun, 02 Jul 2023 00:00:00 GMT    2.0   
3  Tue, 11 Jul 2023 00:00:00 GMT  Tue, 11 Jul 2023 00:00:00 GMT    9.0   
4  Tue, 11 Jul 2023 00:00:00 GMT  Wed, 12 Jul 2023 00:00:00 GMT   14.0   

   C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NAME  \
0       1.0        1.0   5.729125      NaN          NaN   
1       1.0        1.0  13.602380      NaN          NaN   
2       1.0        1.0  27.814460      NaN          NaN   
3       1.0        1.0  58.760230      NaN          NaN   
4       1.0        1.0  70.979000      NaN          NaN   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}      NaN        NaN   2020.0   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}      NaN        NaN   2020.0   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}      NaN        NaN   2020.0   
3  {CB41DB0A-E4B1-489D-A4EA-738F2CD6DB3B}      NaN        NaN   2020.0   
4  {F83F70A4-07A7-40B8-BD51-10CCC1C30D63}      NaN        NaN   2020.0   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...  
3  POLYGON ((-13468077.000 4642260.000, -13467975...  
4  POLYGON ((-13468418.000 4614853.000, -13468428...

In [5]:
# Check the coordinate reference system
print('fire_perimeters CRS: ', fire_perimeters.crs)

# Check whether it is projected or geographic
fire_perimeters.crs

fire_perimeters CRS:  epsg:3857


<Projected CRS: EPSG:3857>
Name: WGS 84 / Pseudo-Mercator
Axis Info [cartesian]:
- X[east]: Easting (metre)
- Y[north]: Northing (metre)
Area of Use:
- name: World between 85.06°S and 85.06°N.
- bounds: (-180.0, -85.06, 180.0, 85.06)
Coordinate Operation:
- name: Popular Visualisation Pseudo-Mercator
- method: Popular Visualisation Pseudo Mercator
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

**Answer to Question 2.)**

- The fire_perimeters data I downloaded from the CalFire website has 22261 observations of 20 columns, which show the perimeters of various historic fires across California, as well as information like the date of the fire, the location, sampling method, etc. 

- The coordinate reference system for this data is Projected: EPSG:3857. 